cmd/kubeadm/app/cmd/init.go

```diff
@@ -233,14 +234,19 @@ func (i *Init) Run(out io.Writer) error {
 		return fmt.Errorf("couldn't parse kubernetes version %q: %v", i.cfg.KubernetesVersion, err)
 	}
 
-	// PHASE 1: Generate certificates
-	if err := cmdphases.CreatePKIAssets(i.cfg); err != nil {
-		return err
-	}
+	if !certphase.UsingExternalCA(i.cfg) {
+		// PHASE 1: Generate certificates
+		if err := cmdphases.CreatePKIAssets(i.cfg); err != nil {
+			return err
+		}
 
-	// PHASE 2: Generate kubeconfig files for the admin and the kubelet
-	if err := kubeconfigphase.CreateInitKubeConfigFiles(kubeadmconstants.KubernetesDir, i.cfg); err != nil {
-		return err
+		// PHASE 2: Generate kubeconfig files for the admin and the kubelet
+		if err := kubeconfigphase.CreateInitKubeConfigFiles(kubeadmconstants.KubernetesDir, i.cfg); err != nil {
+			return err
+		}
+
+	} else {
+		fmt.Printf("[externalca] No ca.key detected, assuming external CA.  Skipping certs and kubeconfig.\n")
```

Comment:
nit: `fmt.Println`
nit: `No ca.key detected, but all other files generated from it are available, so using external CA mode. Won't generate certs nor kubeconfig files` (or something like that)